LinkageCheckerMainIntegrationTest.testBom fails in Java 11

```
$ mvn test -Dtest=LinkageCheckerMainIntegrationTest#testBom
...
[ERROR] Tests run: 5, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 27.32 s <<< FAILURE! - in com.google.cloud.tools.opensource.classpath.LinkageCheckerMainIntegrationTest
[ERROR] testBom(com.google.cloud.tools.opensource.classpath.LinkageCheckerMainIntegrationTest)  Time elapsed: 21.828 s  <<< FAILURE!
org.junit.ComparisonFailure: expected:<Found 8[01] linkage errors> but was:<Found 8[23] linkage errors>
	at com.google.cloud.tools.opensource.classpath.LinkageCheckerMainIntegrationTest.testBom(LinkageCheckerMainIntegrationTest.java:134)
```

Comparing the dashboards build by Java 8 and Java 11 locally.

The class is available in Java 8 but not working on Java 11.

Source code: https://gist.github.com/suztomo/10f3a1afffa0caf555ac0d870211afeb

```
suztomo-macbookpro44% java8
suztomo-macbookpro44% javac suztomo/Main.java
suztomo-macbookpro44% java suztomo/Main
DataSource interface available: Empty Data Source
suztomo-macbookpro44% java suztomo.Main
DataSource interface available: Empty Data Source
suztomo-macbookpro44% java11
suztomo-macbookpro44% java suztomo.Main
Error: Unable to initialize main class suztomo.Main
Caused by: java.lang.NoClassDefFoundError: javax/activation/DataSource
suztomo-macbookpro44% 
```

--- a/dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerMainIntegrationTest.java
+++ b/dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerMainIntegrationTest.java
@@ -30,7 +30,9 @@ import java.nio.file.Path;
 import javax.xml.stream.XMLStreamException;
 import javax.xml.transform.TransformerException;
 import org.eclipse.aether.RepositoryException;
+import org.hamcrest.core.StringStartsWith;
 import org.junit.After;
+import org.junit.Assume;
 import org.junit.Before;
 import org.junit.Test;
 
@@ -125,8 +127,11 @@ public class LinkageCheckerMainIntegrationTest {
   }
 
   @Test
-  public void testBom()
+  public void testBom_java8()
       throws IOException, RepositoryException, TransformerException, XMLStreamException {
+    // The number of linkage errors differ between Java 8 and Java 11 runtime.
+    Assume.assumeThat(System.getProperty("java.version"), StringStartsWith.startsWith("1.8.0"));
+
     try {
       LinkageCheckerMain.main(new String[] {"-b", "com.google.cloud:libraries-bom:1.0.0"});
       fail("LinkageCheckerMain should throw LinkageCheckResultException upon errors");
@@ -142,6 +147,45 @@ public class LinkageCheckerMainIntegrationTest {
             "Class com.google.net.rpc3.client.RpcStubDescriptor is not found;\n"
                 + "  referenced by 21 class files\n"
                 + "    com.google.appengine.api.appidentity.AppIdentityServicePb"
+                + " (com.google.appengine:appengine-api-1.0-sdk:1.9.71)");
+
+    // Show the dependency path to the problematic artifact
+    Truth.assertThat(output)
+        .contains(
+            "com.google.appengine:appengine-api-1.0-sdk:1.9.71 is at:\n"
+                + "  com.google.http-client:google-http-client-appengine:1.29.1 (compile) "
+                + "/ com.google.appengine:appengine-api-1.0-sdk:1.9.71 (provided)");
+  }
+
+  @Test
+  public void testBom_java11()
+      throws IOException, RepositoryException, TransformerException, XMLStreamException {
+    // The number of linkage errors differ between Java 8 and Java 11 runtime.
+    Assume.assumeThat(System.getProperty("java.version"), StringStartsWith.startsWith("11."));
+
+    try {
+      LinkageCheckerMain.main(new String[] {"-b", "com.google.cloud:libraries-bom:1.0.0"});
+      fail("LinkageCheckerMain should throw LinkageCheckResultException upon errors");
+    } catch (LinkageCheckResultException expected) {
+      assertEquals("Found 823 linkage errors", expected.getMessage());
+    }
+
+    String output = readCapturedStdout();
+
+    // Appengine-api-sdk is known to have invalid references.
+    Truth.assertThat(output)
+        .contains(
+            "Class com.google.net.rpc3.client.RpcStubDescriptor is not found;\n"
+                + "  referenced by 21 class files\n"
+                + "    com.google.appengine.api.appidentity.AppIdentityServicePb"
+                + " (com.google.appengine:appengine-api-1.0-sdk:1.9.71)");
+
+    // javax.activation.DataSource has been removed in Java 11.
+    Truth.assertThat(output)
+        .contains(
+            "Class javax.activation.DataSource is not found;\n"
+                + "  referenced by 9 class files\n"
+                + "    com.google.appengine.api.utils.HttpRequestParser"
                 + " (com.google.appengine:appengine-api-1.0-sdk:1.9.71)");
 
     // Show the dependency path to the problematic artifact

--- a/dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerMainIntegrationTest.java
+++ b/dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerMainIntegrationTest.java
@@ -130,7 +130,7 @@ public class LinkageCheckerMainIntegrationTest {
   public void testBom_java8()
       throws IOException, RepositoryException, TransformerException, XMLStreamException {
     // The number of linkage errors differ between Java 8 and Java 11 runtime.
-    Assume.assumeThat(System.getProperty("java.version"), StringStartsWith.startsWith("1.8.0"));
+    Assume.assumeThat(System.getProperty("java.version"), StringStartsWith.startsWith("1.8."));
 
     try {
       LinkageCheckerMain.main(new String[] {"-b", "com.google.cloud:libraries-bom:1.0.0"});
@@ -160,8 +160,10 @@ public class LinkageCheckerMainIntegrationTest {
   @Test
   public void testBom_java11()
       throws IOException, RepositoryException, TransformerException, XMLStreamException {
-    // The number of linkage errors differ between Java 8 and Java 11 runtime.
-    Assume.assumeThat(System.getProperty("java.version"), StringStartsWith.startsWith("11."));
+    // The number of linkage errors differs between Java 8 and Java 11 runtime.
+    String javaVersion = System.getProperty("java.version");
+    int javaMajorVersion = Integer.parseInt(javaVersion.split("\\.")[0]);
+    Assume.assumeTrue(javaMajorVersion >= 11);
 
     try {
       LinkageCheckerMain.main(new String[] {"-b", "com.google.cloud:libraries-bom:1.0.0"});

--- a/dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerMainIntegrationTest.java
+++ b/dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerMainIntegrationTest.java
@@ -162,6 +162,7 @@ public class LinkageCheckerMainIntegrationTest {
       throws IOException, RepositoryException, TransformerException, XMLStreamException {
     // The number of linkage errors differs between Java 8 and Java 11 runtime.
     String javaVersion = System.getProperty("java.version");
+    // javaMajorVersion is 1 when we use Java 8. Still good indicator to ensure Java 11 or higher.
     int javaMajorVersion = Integer.parseInt(javaVersion.split("\\.")[0]);
     Assume.assumeTrue(javaMajorVersion >= 11);
 

